kernel/src/listed_log_files.rs

```diff
@@ -621,4 +636,51 @@ mod list_log_files_with_log_tail_tests {
         assert_source(&result[1], CommitSource::Catalog);
         assert_source(&result[2], CommitSource::Catalog);
     }
+
+    #[test]
+    fn test_listing_omits_staged_commits() {
+        // note that in the presence of staged commits, we CANNOT trust listing to determine which
+        // to include in our listing/log segment. This is up to the catalog. (e.g. version
+        // 5.uuid1.json and 5.uuid2.json can both exist and only catalog can say which is the 'real'
+        // version 5).
+
+        let log_files = vec![
+            (0, LogPathFileType::Commit, CommitSource::Filesystem),
+            (1, LogPathFileType::Commit, CommitSource::Filesystem),
+            (2, LogPathFileType::StagedCommit, CommitSource::Filesystem),
+            (2, LogPathFileType::StagedCommit, CommitSource::Filesystem),
```

Comment:
the duplicate here doesn't do anything. Perhaps change it to 1 to ensure that the commit isn't replaced/dropped by the staged commit.
```suggestion
            (1, LogPathFileType::StagedCommit, CommitSource::Filesystem),
            (2, LogPathFileType::StagedCommit, CommitSource::Filesystem),
```